### PR TITLE
[IMP] mail: create kanban and list activity field components

### PR DESCRIPTION
**Pull request by seb-odoo**
Enterprise: https://github.com/odoo/enterprise/pull/31844

### CODE REVIEW

File: addons/mail/static/src/models/list_field_activity_view.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                    return this.env._t("Warning");
+                }
+                if (this.webRecord.data.activity_summary) {
+                    return this.webRecord.data.activity_summary;
+                }
+                if (this.webRecord.data.activity_type_id) {
+                    return this.webRecord.data.activity_type_id[1]; // 1 = display_name
+                }
+                return '';
+            },
```
Comment (alexkuhn) [inline addons/mail/static/src/models/list_field_activity_view.js:27]: Could `default: ''` so it's just `clear()`

File: addons/mail/static/src/models/web_client_view_activity_button_view.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    },
+    fields: {
+        activityListPopoverView: one('PopoverView', {
+            inverse: 'webClientViewActivityButtonViewOwnerAsActivityList',
+        }),
+        buttonClass: attr({
+            compute() {
+                if (!this.thread) {
+                    return;
+                }
```
Comment (alexkuhn) [inline addons/mail/static/src/models/web_client_view_activity_button_view.js:30]: `return clear()`

File: addons/mail/static/src/models/activity_list_popover_content_view.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, many, one } from '@mail/model/model_field';
+
+registerModel({
+    name: 'ActivityListPopoverContentView',
```
Comment (alexkuhn) [inline addons/mail/static/src/models/activity_list_popover_content_view.js:7]: Could use `.` notation, here `PopoverView.ActivtyListContentView` for example

**Review by alexkuhn — COMMENTED** [review]
global review
(will look at different commits soon)

File: addons/mail/static/src/js/activity.js [hunk 4]
```diff
@@ -47,6 +47,11 @@ const KanbanActivity = AbstractField.extend({
         this.defaultActivityType = options.activityType;
         this.isActivityViewCell = options.isActivityViewCell;
     },
+    async willStart() {
+        await this._super();
+        await owl.Component.env.services.messaging.modelManager.messagingCreatedPromise;
+        this.messaging = owl.Component.env.services.messaging.modelManager.messaging;
```
Comment (alexkuhn) [inline addons/mail/static/src/js/activity.js:53]: why not just `await .get()`?

File: addons/mail/static/src/model/model_manager.js [hunk 5]
```diff
@@ -95,7 +95,7 @@ export class ModelManager {
          * mode adds more integrity checks and more verbose error messages, at
          * the cost of performance.
          */
-        this.isDebug = false;
+        this.isDebug = true;
```
Comment (alexkuhn) [inline addons/mail/static/src/model/model_manager.js]: it's in a `WIP` commit so ok. Just don't forget to reset to `false` before merge

**Review by alexkuhn — COMMENTED** [review]
up to 2nd wip commit

File: addons/mail/static/src/js/activity.js (hunk 4 above)
Comment (seb-odoo) [reply addons/mail/static/src/js/activity.js:53]: get includes messaging initialized which is longer than necessary, just messaging created is enough here

File: addons/mail/static/src/models/activity_list_popover_content_view.js (hunk 3 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/activity_list_popover_content_view.js]: I have shortened all the names.

File: addons/mail/static/src/models/activity_list_popover_content_activity_view.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+import { auto_str_to_date } from 'web.time';
+import { sprintf } from '@web/core/utils/strings';
+
+registerModel({
+    name: 'ActivityListPopoverContentActivityView',
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/static/src/models/activity_list_popover_content_activity_view.js]: shortened names

File: addons/mail/static/src/backend_components/web_client_view_activity_button_view/web_client_view_activity_button_view.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.o_WebClientViewActivityButtonView_buttonDefault {
+    color: #dddddd;
```
[the remark this answers is not in the document]
Comment (seb-odoo) [reply addons/mail/static/src/backend_components/web_client_view_activity_button_view/web_client_view_activity_button_view.scss]: changed to text-muted

File: addons/mail/static/src/models/web_client_view_activity_button_view.js (hunk 2 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/web_client_view_activity_button_view.js]: not really necessary, thread is required this is a purely technical guard (order of compute)

File: addons/mail/static/src/models/list_field_activity_view.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/list_field_activity_view.js]: changed but I'm not a big fan

**Comment by seb-odoo** [comment]
@robodoo override=ci/security

Code move, and injected `$date` is manually built just above from a single `div` of which content is set with `text()`.

File: addons/mail/static/src/models/activity_button_view.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    name: 'ActivityButtonView',
+    identifyingMode: 'xor',
+    recordMethods: {
+        onClick(ev) {
+            if (this.listFieldActivityViewOwner) {
+                ev.stopPropagation(); // prevents list view click that opens form view. TODO use special_click instead?
+            }
+            if (!this.exists()) {
+                return;
+            }
```
Comment (alexkuhn) [inline addons/mail/static/src/models/activity_button_view.js:17]: `this.listFieldActivityViewOwner` will crash before the purpose of `!this.exists()` below.
Do you need to `stopPropagation` before handling `!this.exists()`?

**Review by alexkuhn — APPROVED** [review]
LGTM otherwise

File: addons/mail/static/src/models/activity_button_view.js (hunk 8 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/activity_button_view.js]: I will move it up, even if I needed it it's impossible to code it with a view model alone.

**Comment by alexkuhn** [comment]
@robodoo r+

**Pull request closed without merging**